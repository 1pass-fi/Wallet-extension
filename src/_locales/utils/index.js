export default async function getFetchWithTimeout(url, opts, timeout = 5000) {
  const abortController = new window.AbortController()
  const { signal } = abortController
  const f = window.fetch(url, {
    ...opts,
    signal
  })

  const timer = setTimeout(() => abortController.abort(), timeout)

  try {
    const res = await f
    clearTimeout(timer)
    return res
  } catch (e) {
    clearTimeout(timer)
    throw e
  }
}
