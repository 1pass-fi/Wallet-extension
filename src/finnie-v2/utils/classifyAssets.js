import filter from 'lodash/filter'

const classifyAssets = (assets, collections) => {
  try {
    // add collection name to nfts
    return assets.map(asset => {
      const collectionNames = collections.filter(collection => collection.collection?.includes(asset.txId)).map(collection => collection.name)

      asset.collection = collectionNames ? collectionNames : []
      return asset
    })
  } catch (error) {
    console.log('Fail to classify assets: ', error)
    return assets
  }
}

export default classifyAssets