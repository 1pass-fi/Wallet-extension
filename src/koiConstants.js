export const LOAD_KOI_BY = {
  ADDRESS: 'address',
  KEY: 'key',
}

export const HEADER_EXCLUDE_PATH = [
  '/account/login',
  '/account/login/phrase',
  '/account/connect-site',
  '/account/sign-transaction',
  '/account/welcome',
  '/account/create/success',
  '/account/import/keyfile/success',
  '/account/import/phrase/success'
]
export const NAVBAR_EXCLUDE_PATH = [
  '/account/login',
  '/account/login/phrase',
  '/account/connect-site',
  '/account/sign-transaction',
  '/account/welcome',
  '/account/create',
  '/account/create/success',
  '/account/import/keyfile',
  '/account/import/keyfile/success',
  '/account/import/phrase',
  '/account/import/phrase/success'
]

export const LONG_LIVED_HANDLER = [
  'GET_ADDRESS_SUCCESS',
  'GET_ADDRESS_ERROR',
  'GET_PERMISSION_SUCCESS',
  'GET_PERMISSION_ERROR',
  'CREATE_TRANSACTION_SUCCESS',
  'CREATE_TRANSACTION_ERROR',
  'CONNECT_SUCCESS',
  'CONNECT_ERROR',
  'KOI_GET_ADDRESS_SUCCESS',
  'KOI_GET_ADDRESS_ERROR',
  'KOI_GET_PERMISSION_SUCCESS',
  'KOI_GET_PERMISSION_ERROR',
  'KOI_CREATE_TRANSACTION_SUCCESS',
  'KOI_CREATE_TRANSACTION_ERROR',
  'KOI_CONNECT_SUCCESS',
  'KOI_CONNECT_ERROR',
  'KOI_DISCONNECT_SUCCESS',
  'KOI_DISCONNECT_ERROR',
  'KOI_REGISTER_DATA_SUCCESS',
  'KOI_REGISTER_DATA_ERROR',
  'GET_BALANCES_SUCCESS',
  'GET_ALL_ADDRESSES',
  'GET_ALL_ADDRESSES_SUCCESS',
  'GET_ALL_ADDRESSES_ERROR',
  'GET_WALLET_NAMES',
  'GET_WALLET_NAMES_SUCCESS',
  'GET_WALLET_NAMES_ERROR',
  'DISCONNECT',
  'DISCONNECT_SUCCESS',
  'DISCONNECT_ERROR',
  'GET_PUBLIC_KEY',
  'GET_PUBLIC_KEY_SUCCESS',
  'GET_PUBLIC_KEY_ERROR',
  'ERROR',
  'KOI_SIGN_PORT_SUCCESS',
  'KOI_SIGN_PORT_ERROR',
  'KOI_SEND_KOI_SUCCESS',
  'KOI_SEND_KOI_ERROR',
  'UPLOAD_NFT_SUCCESS'
]

export const PORTS = {
  POPUP: 'POPUP',
  CONTENT_SCRIPT: 'CONTENT_SCRIPT',
}

export const MESSAGES = {
  IMPORT_WALLET: 'IMPORT_WALLET',
  IMPORT_WALLET_SUCCESS: 'IMPORT_WALLET_SUCCESS',
  GET_BALANCES: 'GET_BALANCES',
  GET_BALANCES_SUCCESS: 'GET_BALANCES_SUCCESS',
  LOAD_WALLET: 'LOAD_WALLET',
  LOAD_WALLET_SUCCESS: 'LOAD_WALLET_SUCCESS',
  REMOVE_WALLET: 'REMOVE_WALLET',
  REMOVE_WALLET_SUCCESS: 'REMOVE_WALLET_SUCCESS',
  LOCK_WALLET: 'LOCK_WALLET',
  LOCK_WALLET_SUCCESS: 'LOCK_WALLET_SUCCESS',
  UNLOCK_WALLET: 'UNLOCK_WALLET',
  UNLOCK_WALLET_SUCCESS: 'UNLOCK_WALLET_SUCCESS',
  GENERATE_WALLET: 'GENERATE_WALLET',
  GENERATE_WALLET_SUCCESS: 'GENERATE_WALLET_SUCCESS',
  SAVE_WALLET: 'SAVE_WALLET',
  SAVE_WALLET_SUCCESS: 'SAVE_WALLET_SUCCESS',
  LOAD_CONTENT: 'LOAD_CONTENT',
  LOAD_CONTENT_SUCCESS: 'LOAD_CONTENT_SUCCESS',
  LOAD_ACTIVITIES: 'LOAD_ACTIVITIES',
  LOAD_ACTIVITIES_SUCCESS: 'LOAD_ACTIVITIES_SUCCESS',
  ERROR: 'ERROR',
  MAKE_TRANSFER: 'MAKE_TRANSFER',
  MAKE_TRANSFER_SUCCESS: 'MAKE_TRANSFER_SUCCESS',
  GET_ADDRESS: 'GET_ADDRESS',
  GET_ADDRESS_SUCCESS: 'GET_ADDRESS_SUCCESS',
  GET_ADDRESS_ERROR: 'GET_ADDRESS_ERROR',
  GET_PERMISSION: 'GET_PERMISSION',
  GET_PERMISSION_SUCCESS: 'GET_PERMISSION_SUCCESS',
  GET_PERMISSION_ERROR: 'GET_PERMISSION_ERROR',
  CONNECT: 'CONNECT',
  CONNECT_SUCCESS: 'CONNECT_SUCCESS',
  CONNECT_ERROR: 'CONNECT_ERROR',
  GET_KEY_FILE: 'GET_KEY_FILE',
  GET_KEY_FILE_SUCCESS: 'GET_KEY_FILE_SUCCESS',
  CREATE_TRANSACTION: 'CREATE_TRANSACTION',
  CREATE_TRANSACTION_SUCCESS: 'CREATE_TRANSACTION_SUCCESS',
  CREATE_TRANSACTION_ERROR: 'CREATE_TRANSACTION_ERROR',
  SIGN_TRANSACTION: 'SIGN_TRANSACTION',
  SIGN_TRANSACTION_SUCCESS: 'SIGN_TRANSACTION_SUCCESS',
  SIGN_TRANSACTION_ERROR: 'SIGN_TRANSACTION_ERROR',
  KOI_GET_ADDRESS: 'KOI_GET_ADDRESS',
  KOI_GET_ADDRESS_SUCCESS: 'KOI_GET_ADDRESS_SUCCESS',
  KOI_GET_ADDRESS_ERROR: 'KOI_GET_ADDRESS_ERROR',
  KOI_GET_PERMISSION: 'KOI_GET_PERMISSION',
  KOI_GET_PERMISSION_SUCCESS: 'KOI_GET_PERMISSION_SUCCESS',
  KOI_GET_PERMISSION_ERROR: 'KOI_GET_PERMISSION_ERROR',
  KOI_CREATE_TRANSACTION: 'KOI_CREATE_TRANSACTION',
  KOI_CREATE_TRANSACTION_SUCCESS: 'KOI_CREATE_TRANSACTION_SUCCESS',
  KOI_CREATE_TRANSACTION_ERROR: 'KOI_CREATE_TRANSACTION_ERROR',
  KOI_CONNECT: 'KOI_CONNECT',
  KOI_CONNECT_SUCCESS: 'KOI_CONNECT_SUCCESS',
  KOI_CONNECT_ERROR: 'KOI_CONNECT_ERROR',
  KOI_DISCONNECT: 'KOI_DISCONNECT',
  KOI_DISCONNECT_SUCCESS: 'KOI_DISCONNECT_SUCCESS',
  KOI_DISCONNECT_ERROR: 'KOI_DISCONNECT_ERROR',
  GET_WALLET: 'GET_WALLET',
  GET_WALLET_SUCCESS: 'GET_WALLET_SUCCESS',
  KOI_REGISTER_DATA: 'KOI_REGISTER_DATA',
  KOI_REGISTER_DATA_SUCCESS: 'KOI_REGISTER_DATA_SUCCESS',
  KOI_REGISTER_DATA_ERROR: 'KOI_REGISTERs_DATA_ERROR',
  UPLOAD_NFT: 'UPLOAD_NFT',
  TEST: 'TEST',
  TEST_SUCCESS: 'TEST_SUCCESS',
  CREATE_COLLECTION: 'CREATE_COLLECTION',
  CREATE_KID: 'CREATE_KID',
  UPDATE_KID: 'UPDATE_KID',
  CHANGE_ACCOUNT_NAME: 'CHANGE_ACCOUNT_NAME',
  GET_LOCK_STATE: 'GET_LOCK_STATE',
  LOAD_COLLECTIONS: 'LOAD_COLLECTIONS',
  LOAD_KID: 'LOAD_KID',
  CREATE_UPDATE_KID: 'CREATE_UPDATE_KID',
  LOAD_PENDING_ASSETS: 'LOAD_PENDING_ASSETS',
  GET_ALL_ADDRESSES: 'GET_ALL_ADDRESSES',
  GET_ALL_ADDRESSES_SUCCESS: 'GET_ALL_ADDRESSES_SUCCESS',
  GET_ALL_ADDRESSES_ERROR: 'GET_ALL_ADDRESSES_ERROR',
  GET_WALLET_NAMES: 'GET_WALLET_NAMES',
  GET_WALLET_NAMES_SUCCESS: 'GET_WALLET_NAMES_SUCCESS',
  GET_WALLET_NAMES_ERROR: 'GET_WALLET_NAMES_ERROR',
  DISCONNECT: 'DISCONNECT',
  DISCONNECT_SUCCESS: 'DISCONNECT_SUCCESS',
  DISCONNECT_ERROR: 'DISCONNECT_ERROR',
  GET_PUBLIC_KEY: 'GET_PUBLIC_KEY',
  GET_PUBLIC_KEY_SUCCESS: 'GET_PUBLIC_KEY_SUCCESS',
  GET_PUBLIC_KEY_ERROR: 'GET_PUBLIC_KEY_ERROR',
  ENCRYPT: 'ENCRYPT',
  ENCRYPT_SUCCESS: 'ENCRYPT_SUCCESS',
  ENCRYPT_ERROR: 'ENCRYPT_ERROR',
  KOI_SIGN_PORT: 'KOI_SIGN_PORT',
  KOI_SIGN_PORT_SUCCESS: 'KOI_SIGN_PORT_SUCCESS',
  KOI_SIGN_PORT_ERROR: 'KOI_SIGN_PORT_ERROR',
  KOI_SEND_KOI: 'KOI_SEND_KOI',
  KOI_SEND_KOI_SUCCESS: 'KOI_SEND_KOI_SUCCESS',
  KOI_SEND_KOI_ERROR: 'KOI_SEND_KOI_ERROR',
  SAVE_WALLET_GALLERY: 'SAVE_WALLET_GALLERY'
}

export const PATH = {
  // Router path
  IMPORT_PHRASE_REDIRECT: '/account/import/phrase/success',
  IMPORT_KEY_REDIRECT: '/account/import/keyfile/success',
  CREATE_WALLET_REDIRECT: '/account/create/success',
  HOME: '/account',
  LOGIN: '/account/login',
  ACTIVITY: '/activity',
  WELCOME: '/account/welcome',
  SETTING: '/settings',
  CONNECT_SITE: '/account/connect-site',
  SIGN_TRANSACTION: '/account/sign-transaction',
  // Gallery path
  NFT_IMAGE: 'https://arweave.net',
  VIEW_BLOCK: 'https://viewblock.io/arweave/address',
  ETHERSCAN: 'https://etherscan.io/address',
  KOI_ROCK: 'https://koi.rocks/content-detail',
  GALLERY: chrome.extension
    ? `${chrome.extension.getURL('/options.html')}`
    : 'GALLERY_PATH',
  ALL_CONTENT: 'https://devbundler.openkoi.com:8888/state/top-content-predicted?frequency=all',
  SINGLE_CONTENT: 'https://devbundler.openkoi.com:8888/state/nft?tranxId=',
  VIEW_BLOCK_TRANSACTION: 'https://viewblock.io/arweave/tx',
  // Fetch AR price
  AR_PRICE: 'https://api.coingecko.com/api/v3/simple/price?ids=arweave&vs_currencies=usd',
  // Social sharing
  SHARE_TWITTER: 'https://twitter.com/intent/tweet',
  SHARE_FACEBOOK: 'https://www.facebook.com/sharer/sharer.php',
  SHARE_LINKEDIN: 'https://www.linkedin.com/sharing/share-offsite/',
  // Affiliate endpoints
  AFFILIATE_REGISTER: 'https://koi.rocks:8888/api/v1/registerAffiliate',
  AFFILIATE_CLAIM_REWARD: 'https://koi.rocks:8888/api/v1/cliamReward',
  AFFILIATE_REGISTRATION_REWARD: 'https://koi.rocks:8888/api/v1/freeRegistrationReward',
  AFFILIATE_SUBMIT_CODE: 'https://koi.rocks:8888/api/v1/submitCode',
  AFFILIATE_TOTAL_REWARD: 'https://koi.rocks:8888/api/v1/totalRewardForAffiliate'
}

export const ERROR_MESSAGE = {
  INCORRECT_PASSWORD: 'Incorrect password, please try again.',
  PASSWORD_LENGTH: 'Your Koii password must have at least 8 characters.',
  PASSWORD_MATCH: 'Passwords do not match, please try again.',
  INCORRECT_PHRASE: 'Incorrect seed phrase, please try again.',
  EMPTY_PHRASE: 'Seed phrase should not be blank.',
  CHECKED_TERMS: 'You have to agree to the Terms of Service',
  EMPTY_FIELDS: 'Please fill in all fields.',
  NOT_ENOUGH_KOI: `You don't have enough Koii.`,
  REQUEST_NOT_EXIST: 'This request is no longer exist.',
  INVALID_AMOUNT: 'Amount can not be negative.',
  ALREADY_CONNECTED_SITE: 'This site has already connected.',
  MUST_USE_IN_POPUP: 'Please use this feature in the extension pop up.',
  NOT_ENOUGH_AR: `You don't have enough AR.`,
  FILE_TOO_LARGE: 'File too large. The maximum size for NFT is 15MB',
  CANNOT_GET_COSTS: 'Cannot get the upload costs.',
  INVALID_FRIEND_CODE: `That code isn't quite right. Check it and try again.`,
  COLLECTION_NFT_EMPTY: 'A collection has to contain at least one NFT.',
  KID_FILE_TOO_LARGE: 'File too large. The maximum size for Profile Picture is 500KB',
  PASSWORD_REQUIRED: 'Password required.',
  SELECT_ACCOUNT: 'Please select account.',
  SELECT_TOKEN: 'Please select token.',
  CREATE_WALLET_FAILED: 'Create new wallet failed.'
}

export const STORAGE = {
  KOI_ADDRESS: 'koiAddress',
  KOI_KEY: 'koiKey',
  KOI_BALANCE: 'koiBalance',
  AR_BALANCE: 'arBalance',
  CONTENT_LIST: 'contentList',
  ACTIVITIES_LIST: 'activitiesList',
  SITE_PERMISSION: 'sitePermission',
  PENDING_REQUEST: 'pendingRequest',
  KOI_PHRASE: 'koiPhrase',
  PENDING_TRANSACTION: 'pendingTransactions',
  ACCOUNT_NAME: 'accountName',
  PRICE: 'PRICE',
  AFFILIATE_CODE: 'AFFILIATE_CODE',
  SHOW_WELCOME_SCREEN: 'SHOW_WELCOME_SCREEN',
  CURRENCY: 'CURRENCY',
  MOCK_COLLECTIONS_STORE: 'MOCK_COLLECTIONS_STORE',
  SHOW_VIEWS: 'SHOW_VIEWS',
  SHOW_EARNED_KOI: 'SHOW_EARNED_KOI',
  COLLECTIONS: 'COLLECTIONS',
  KID: 'KID'
}

export const REQUEST = {
  PERMISSION: 'PERMISSION',
  TRANSACTION: 'TRANSACTION',
}

export const RATE = {
  KOII: 0.93,
  AR: 2,
  ETH: 100
}

export const NOTIFICATION = {
  COPIED: 'Copied to clipboard.',
  KEY_EXPORTED: 'Private key downloaded.',
  ACCOUNT_NAME_UPDATED: 'Account name updated.',
  NFT_LOADED: 'Your assets list is up to date.',
  TRANSACTION_SENT: 'Transaction sent.',
  CREATE_COLLECTION_SUCCESS: 'Collection created successfully. It may take a while until you can get your data updated.',
  CREATE_KID_SUCCESS: 'Created KID successfully. It may take a while until you can get your data updated.',
  UPDATE_KID_SUCCESS: 'Updated KID successfully. It may take a while until you can get your data updated.'
}

export const WARNING_MESSAGE = {
  SEND_ZERO_KOI: 'You are about to send 0 tokens. There will still be a very small fee. Make sure the transaction is correct before finalizing it.',
}

export const STATEMENT = {
  NO_REWARD: `You don't have any rewards available, share your code to get some!`,
}

export const LOAD_BALANCES_TIME_INTERVAL = 10000

export const NFT_BIT_DATA = 'NFT_BIT_DATA'

export const ALL_NFT_LOADED = 'ALL_NFT_LOADED'

export const DISCONNECTED_BACKGROUND = 'Attempting to use a disconnected port object'

export const TRANSACTION_DATA = 'TRANSACTION_DATA'

export const OS = 'OS'

export const WINDOW_SIZE = {
  WIN_HEIGHT: 635,
  WIN_WIDTH: 439,
  MAC_HEIGHT: 628,
  MAC_WIDTH: 426
}

export const MOCK_COLLECTIONS_STORE = 'MOCK_COLLECTIONS_STORE'

export const FILENAME = {
  ARWEAVE: 'arweave-key.json',
  ETHEREUM: 'ethereum-key.json'
}

export const URL = {
  TERM_OF_SERVICE: 'https://koii.network/TOU_June_22_2021.pdf'
}

export const GALLERY_IMPORT_PATH = ['/welcome', '/upload-wallet', '/import-wallet', '/create-wallet']
