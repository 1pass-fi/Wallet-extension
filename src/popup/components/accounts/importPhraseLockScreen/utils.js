import { PATH, ERROR_MESSAGE } from 'constants'

export const validatePhrase = ({
  phrase,
  password,
  confirmPassword,
  history,
  setError,
  importWallet,
}) => {
  try {
    if (phrase.split(' ').length != 12) {
      setError(ERROR_MESSAGE.INCORRECT_PHRASE)
    } else if (password.length < 8) {
      setError(ERROR_MESSAGE.PASSWORD_LENGTH)
    } else if (password !== confirmPassword) {
      setError(ERROR_MESSAGE.PASSWORD_MATCH)
    } else {
      const redirectPath = PATH.IMPORT_KEY_REDIRECT
      importWallet({ data: phrase, password, history, redirectPath })
    }
  } catch (err) {
    /* istanbul ignore next */
    setError(err.message)
  }
}
