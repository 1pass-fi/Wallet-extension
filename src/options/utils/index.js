import { Web } from '@_koi/sdk/web'
import { MOCK_COLLECTIONS_STORE, STORAGE } from 'constants/koiConstants'
import { isString } from 'lodash'
import { find, get } from 'lodash'
import numeral from 'numeral'
import { getChromeStorage, setChromeStorage } from 'utils'

export const koi = new Web()

import storage from 'services/storage'

export const formatNumber = (value, decimal) => {
  const zeroArray = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
  return numeral(value).format(
    decimal ? `0,0.${zeroArray.slice(0, decimal).join('')}` : '0,0.00'
  )
}

export const getDisplayAddress = (address, head = 6, tail = 4) => {
  try {
    if (isString(address)) return `${address.slice(0, head)}...${address.slice(address.length - tail)}`
  } catch (err) {
    throw new Error(err.message)
  }
}

/**
 * 
 * @param {File} file file object
 */
export const saveImageDataToStorage = async (file) => {
  const url = URL.createObjectURL(file)
  // console.log('bottomButton- file', file)

  // get arrayBuffer
  const response = await fetch(url)
  const blob = await response.blob()
  const dataBuffer = await blob.arrayBuffer()
  // console.log('bottomButton- dataBuffer', dataBuffer)

  // create a 8bit array and save to local storage
  let u8 = new Int8Array(dataBuffer)
  // console.log('bottomButton- u8', u8)

  // save u8 to local storage
  u8 = JSON.stringify(u8, null, 2)
  // console.log('bottomButton- u8', u8)
  await setChromeStorage({ NFT_BIT_DATA: u8 })
  await storage.generic.set.nftBitData(u8)
}

/**
 *
 * @param {Array} collection { tx: [id1, id2] } 
 */
export const mockSaveCollections = async (collection) => {
  console.log('new collection: ', collection)
  const storage = await getChromeStorage(MOCK_COLLECTIONS_STORE)
  const collections = storage[MOCK_COLLECTIONS_STORE] || []
  collections.push(collection)

  await setChromeStorage({[MOCK_COLLECTIONS_STORE]: collections})
}

export const mockGetCollections = async () => {
  const storage = await getChromeStorage(MOCK_COLLECTIONS_STORE)
  const collections = storage[MOCK_COLLECTIONS_STORE] || []
  return collections
}

export const stringTruncate = (str, length) => {
  try {
    if (str.length > 20) {
      return `${str.slice(0,length)}...`
    }
    return str
  } catch (err) {
    console.log(err)
  }
}

export const JSONFileToObject = async (file) => {
  try {
    const fileText = await file.text()
    return JSON.parse(fileText)
  } catch (err) {
    throw new Error(err.message)
  }
}
