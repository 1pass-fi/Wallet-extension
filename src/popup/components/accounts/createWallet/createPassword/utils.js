import get from 'lodash/get'
import { ERROR_MESSAGE } from 'koiConstants'

export const validatePassword = ({ e, setError, generateWallet }) => {
  try {
    const pwd = get(e, 'target.pwd.value')
    const pwdConfirm = get(e, 'target.pwdConfirm.value')
    const checked = get(e, 'target.checkbox.checked')
    if (pwd.length < 8) {
      setError(ERROR_MESSAGE.PASSWORD_LENGTH)
    } else if (pwd !== pwdConfirm) {
      setError(ERROR_MESSAGE.PASSWORD_MATCH)
    } else if (!checked) {
      setError(ERROR_MESSAGE.CHECKED_TERMS)
    } else {
      generateWallet({ stage: 2, password: pwd })
    }
  } catch (err) {
    setError(err.message)
  }
}
