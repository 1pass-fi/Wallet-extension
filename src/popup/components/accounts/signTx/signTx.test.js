import '@babel/polyfill'
import { fireEvent, getByText, queryAllByTestId, render, wait, waitFor } from '@testing-library/react'
import { renderHook } from '@testing-library/react-hooks'
import '@testing-library/jest-dom/extend-expect'
import React from 'react'
import sinon from 'sinon'

import { utils } from 'utils'
import { SignTx } from '.'
import { STORAGE } from 'koiConstants'


describe('Test for ImportPhraseLockScreen component', () => {
  let setErrorMock, signTransactionMock, container, accountName, price, getChromeStorageStub, sourceAddressEle, targetAddressEle, originEle
  beforeEach(async () => {
    getChromeStorageStub = sinon.stub(utils, 'getChromeStorage')
    getChromeStorageStub.withArgs(STORAGE.PENDING_REQUEST).resolves({ [STORAGE.PENDING_REQUEST]: { data: {
      origin: 'origin',
      qty: 100,
      address: 'targetAddress'
    } } })
    getChromeStorageStub.withArgs(STORAGE.KOI_ADDRESS).resolves({ [STORAGE.KOI_ADDRESS]: 'koiAddress' })
    setErrorMock = jest.fn()
    signTransactionMock = jest.fn()
    accountName = 'accountName',
    price = { AR: 1, KOI: 2 }

    container = render(<SignTx
      signTransaction={signTransactionMock}
      setError={setErrorMock}
      accountName={accountName}
      price={price}
    />).container
    await waitFor(() => {
      sourceAddressEle = queryAllByTestId(container, 'source-address')[0]
      targetAddressEle = queryAllByTestId(container, 'target-address')[0]
      originEle = queryAllByTestId(container, 'origin')[0]
    })

  })

  afterEach(() => {
    sinon.reset()
    getChromeStorageStub.restore()
  })

  describe('renders without crashing', () => {
    it('renders correctly', async () => {
      expect(container).toMatchSnapshot()
    })
  })

  describe('load data from local storage and set state', () => {
    it ('renders with data from local storage', () => {
      expect(sourceAddressEle.textContent).toEqual('koiAddress')
      expect(targetAddressEle.textContent).toEqual('targetAddress')
      expect(originEle.textContent).toEqual('origin')
    })
  })
})
