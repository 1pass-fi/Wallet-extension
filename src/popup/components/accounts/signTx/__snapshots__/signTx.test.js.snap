// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for ImportPhraseLockScreen component renders without crashing renders correctly 1`] = `
<div>
  <div
    class="sign-tx"
  >
    <div
      class="account-section"
    >
      <div
        class="account-source"
      >
        <div>
          From:
        </div>
        <div
          class="source-account account"
        >
          <div
            class="logo-icon"
          >
            <svg
              class="icon"
            />
          </div>
          <div
            class="account-info"
          >
            <div
              class="title"
            >
              Account 1
            </div>
            <div
              class="address"
            />
          </div>
        </div>
        <div>
          To:
        </div>
        <div
          class="destination-account account"
        >
          <div
            class="logo-icon"
          >
            <svg />
          </div>
          <div
            class="account-info"
          >
            <div
              class="address"
            />
          </div>
        </div>
      </div>
      <div
        class="origin"
      >
        <a
          href=""
        />
      </div>
    </div>
    <div
      class="content-section"
    >
      <div
        class="card transaction-detail"
      >
        <div
          class="label"
        >
          Transaction details
        </div>
        <div
          class="details"
        >
          <div
            class="detail-row"
          >
            Send
          </div>
          <div
            class="detail-row amount"
          >
            <div
              class="koi"
            >
              ---
               KOI
            </div>
            <div
              class="usd"
            >
              ~
              0
               USD
            </div>
          </div>
        </div>
        <div
          class="button-group"
        >
          <button
            class="button-shared button-filled button confirm"
          >
            Confirm
          </button>
          <button
            class="button-shared button-layout button reject"
          >
            Reject
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
